doc/decisions/DR024-Versioning-traits-and-specifications-method.md

```diff
@@ -0,0 +1,365 @@
+  # contain the locatableContent trait.
+  locatableContent_version = locatableContent_version_from_trait_set(traitSet)
+
+  success_result = TraitsData()
+  if (locatableContent_version == 2):
+    trait = mediacreation.traits.content.LocatableContentTrait(success_result)
+    trait.setUrl(managed_filesystem_locations[ref.toString()])
+    successCallback(idx, success_result)
+  elif (locatableContent_version == 1):
+    trait = mediacreation.legacy.traits.content.LocatableContentTrait_v1(success_result)
+    trait.setLocation(managed_filesystem_locations[ref.toString()])
+    successCallback(idx, success_result)
+  else:
+    raise RuntimeError()
+
+```
+
+#### Pros
+- Both managers and hosts have the option to be backwards compatible.
+- More flexible, traits of different versions can be mixed in the same
+  session, which could allow hosts/managers to perform partial upgrades
+  in the case they are blocked on small specifics.
+- Hosts don't have to interrogate and declare which library versions
+  are in use, which helps especially in the case of second-order
+  dependencies.
+- Seems to fit with existing OpenAssetIO design, especially with
+  `BatchElementError`, as an incompatible version wouldn't necessarily
+  need to error the whole batch.
+- No new API, core `openassetio` library remains version unaware.
+- Cannot be ignored, unlike other "layered on top" mechanisms.
+
+#### Cons
+- No "primary" version across the session, increasing implementation
+  complexity.
+- Introduces "Try and see" negotiation on the host side, although
+  introspection behaviour to negate this could be introduced at a later
+  date, similar to other introspection mechanisms. (See Option 3 -
+  Extended)
+- Would likely require some utilities in `traitgen`, otherwise, due to
+  each new version effectively being a brand new trait, constructing the
+  correct trait view would become a tedious dance.
+- EntityTraits becomes more complex, as we either have to declare it
+  version agnostic somehow, or managers would need to return the correct
+  versions/entire swathes of versions.
+- Unclear how specification versioning is to work without an out-of-data
+  version.
+
+### In data - Extended
+The implementation of in-data versioning is very basic, which is part of
+its strength, but it necessitates a "try and see" approach on the part
+of the host. If the host wants to be adaptive to the manager, it may
+benefit from a more up front introspection of the supported manager
+versions.
+
+To this end, the `supportedSchemaVersions` method (or similar concept,)
+from Option 2 could be orthogonally added without any conceptual
+conflict. This would have the added benefit of allowing hosts to also
+support a range of version/have backwards compatibility, which is a huge
+benefit.
+
+This stops short of negotiation, being merely introspection similar to
+our other manager introspection methods.
+
+### Reconsidering Schema Version
```

Comment:
[consider] I think this section could be part of the "Relevant data"

Reply:
Aye, fair enough, moved it : 21688f11635f47e872d6976902f14a65af81d2da